File: Assets/Scripts/Client/dto/PlayerEndRoundDto.cs
namespace client.dto
{
    public class PlayerEndRoundDto : IDto
    {
        public string PlayerId { get; private set; }
        public PlayerFaction Faction { get; private set; } 
        public byte PlayerCardId { get; private set; }
        public byte PlayerScore { get; private set; }
        public byte NewCardId { get; private set; }
        
        public PlayerEndRoundDto(CustomMemoryStream ms)
        {
            ReadFromStream(ms);
        }
        
        public void WriteToStream(CustomMemoryStream ms)
        {
        }

        public void ReadFromStream(CustomMemoryStream ms)
        {
            PlayerId = ms.ReadString();
            Faction = (PlayerFaction)ms.ReadByte();
            PlayerCardId = ms.ReadByte();
            PlayerScore = ms.ReadByte();
            NewCardId = ms.ReadByte();
        }
    }
}

File: server/dto/PlayerEndRoundDto.js
export class PlayerEndRoundDto
{
    playerId;
    color;
    playerCardId;
    playerScore;
    newCardId;
    
    constructor(playerId, color, playerCardId, playerScore, newCardId){
        this.playerId = playerId;
        this.color = color;
        this.playerCardId = playerCardId;
        this.playerScore = playerScore;
        this.newCardId = newCardId;
    }

    writeToBuffer(buffer)
    {
        let length = this.playerId.length;
        buffer.writeUInt8(length);
        buffer.writeString(this.playerId);
        
        buffer.writeUInt8(this.color);
        
        buffer.writeUInt8(this.playerCardId);
        buffer.writeUInt8(this.playerScore);
        buffer.writeUInt8(this.newCardId);
    }
}